gtest/src/manager.rs

```diff
@@ -546,3 +560,44 @@ impl JournalHandler for ExtManager {
         // todo!() #714
     }
 }
+
+#[derive(Debug)]
+pub struct Mailbox{
+    mail: HashMap<MessageId, Message>
+}
+
+impl Mailbox {
+    pub fn new(message: Message) -> Mailbox {
+        let mut mailbox = Mailbox{mail: HashMap::new()};
+        mailbox.insert(message);
+        mailbox
+    }
+
+    pub fn get(&self, payload: &Payload) -> Option<&Message> {
+        for message in self.mail.values(){
+            if message.payload.eq(payload){
+                return Option::Some(message);
+            }
+        }
+        None
+    }
+
+    pub fn insert(&mut self, message: Message) -> Option<Message> {
+        self.mail.insert(message.id.clone(), message)
+    }
+
+    pub fn remove_message(&mut self, message_id: &MessageId) -> Result<Message, String>{
+        match self.mail.remove(message_id) {
+            None => {Result::Err(String::from("No message with such message id"))}
+            Some(message) => {Result::Ok(message)}
+        }
```

Comment:
Use `Option::ok_or_else()`:
```suggestion
        self.mail.remove(message_id).ok_or_else(|_| "No message with such message id").to_string())
```